runtime/doc/filetype.txt

```diff
@@ -663,6 +663,19 @@ To disable this behavior, set the following variable in your vimrc: >
 
 	let g:python_recommended_style = 0
 
+QUERY					                *ft-query-plugin*
+
+
+By default enables lints tree-sitter query for installed parsers using
+|lua-treesitter-query_linter|. Query linting can be disabled using >lua
+
+	vim.g.query_lint_on = {}
```

Comment:
Does this need to be a global variable? I think we could achieve the same effect with a buffer local variable. Users can override it either by creating a `.config/nvim/ftplugin/query.lua` file or `autocmd FileType query let b:lint_on = {}`.

Reply:
FileType seems to trigger after the filetype plugin

Reply:
A `FileType` autocommand defined in init.vim/init.lua should trigger before any ftplugins. If a user uses `~/.config/nvim/ftplugin/query.lua` instead, that will also run before the bundled ftplugin.

Reply:
For me `BufReadPre` happens early enough, `BufRead`/`FileType` will not change the filetype when I print in the filetype plugin.